src/java/org/learningequality/FullScreen.java

```diff
@@ -41,37 +44,55 @@ public static FullScreen getInstance() {
         return fullScreen;
     }
 
+    public String getUrl() {
+        return mMainWebView.getUrl();
+    }
+
     public void replaceUrl(String url) {
-        // It is important that we call mWebView.clearHistory after navigation
+        // It is important that we call mMainWebView.clearHistory after navigation
         // is completed, so we will set clearHistoryOnPageFinished and do the
         // work of clearing history in the onPageFinished callback.
         clearHistoryOnPageFinished = true;
-        mWebView.loadUrl(url);
+        mActivity.loadUrl(url);
+    }
+
+    public void showLoadingPage(String loadingUrl) {
+        if (loadingUrl != null && !mLoadingWebView.getUrl().equals(loadingUrl)) {
+            mLoadingWebView.loadUrl(loadingUrl);
+        }
+        mActivity.displayLoadingWebView();
+    }
+
+    public void setAppKeyCookie(String url, String appKey) {
+        CookieManager.getInstance().setCookie(url, "app_key_cookie=" + appKey);
     }
 
     // Configure the WebView to allow fullscreen based on:
     // https://stackoverflow.com/questions/15768837/playing-html5-video-on-fullscreen-in-android-webview/56186877#56186877
     public void configure(final Runnable startWithNetwork, final Runnable startWithUSB, final Runnable loadingReady) {
-        mWebView.setWebContentsDebuggingEnabled(true);
-        mWebView.setWebViewClient(new WebViewClient() {
+        Log.i("Endless Key", "FullScreen configure");
```

Comment:
Nit: Maybe we can make "Endless Key" a constant:
```
private static final String TAG = "Endless Key";

...

Log.i(TAG, "FullScreen configure");

```

Reply:
:+1: